docs/guide/search/extend_elasticsearch.md

```diff
@@ -291,7 +291,210 @@
+                    $ranges[] = [
+                        'from' => $range->getFrom()
+                    ];
+                } else {
+                    // invalid range
+                }
+            }
+
+            return [
+                'range' => [
+                    'field' => 'priority_i',
+                    'ranges' => $ranges
+                ]
+            ];
+        }
+    }
+    ```
+    
+    The `supports()` method checks whether the provided aggregation is of the supported type
+    (in this case, your custom `PriorityRangeAggregation`).
+    
+    The `visit()` method returns an array of results.
+    
+    You also need to create a result extractor, implementing `Ibexa\Platform\Contracts\ElasticSearchEngine\Query\AggregationResultExtractor`,
+    that transforms raw aggregation results from Elasticsearch into `AggregationResult` objects:
+    
+    ``` php
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Query\Aggregation\Elasticsearch;
+
+    use App\Query\Aggregation\PriorityRangeAggregation;
+    use eZ\Publish\API\Repository\Values\Content\Query\Aggregation;
+    use eZ\Publish\API\Repository\Values\Content\Search\AggregationResult;
+    use Ibexa\Platform\Contracts\ElasticSearchEngine\Query\AggregationResultExtractor;
+    use Ibexa\Platform\Contracts\ElasticSearchEngine\Query\LanguageFilter;
+    use stdClass;
+
+    final class PriorityAggregationResultExtractor implements AggregationResultExtractor
+    {
+
+        public function supports(Aggregation $aggregation, LanguageFilter $languageFilter): bool
+        {
+            return $aggregation instanceof PriorityRangeAggregation;
+        }
+
+        public function extract(Aggregation $aggregation, LanguageFilter $languageFilter, array $data): AggregationResult
+        {
+            $entries = [];
+
+            foreach ($data['buckets'] as $bucket) {
+                $entries[] = new AggregationResult\RangeAggregationResultEntry(
+                    new Aggregation\Range($bucket['from'] ?? null, $bucket['to'] ?? null),
+                    $bucket['doc_count']
+                );
+            }
+
+            return new AggregationResult\RangeAggregationResult($aggregation->getName(), $entries);
+        }
+    }
+    ```
+    
+    The `supports()` method checks whether the provided aggregation is of the supported type
+    (in this case, your custom `PriorityRangeAggregation`).
+    
+    The `extract()` method converts the [JSON-encoded data provided by the search engine](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations.html) to a `RangeAggregationResult` object.
+    
+    Finally, register both the aggregation visitor and the result extractor as services.
+    
+    Tag the aggregation visitor with `ezplatform.search.elasticsearch.query.location.aggregation_visitor`
+    and the result extractor with `ezplatform.search.elasticsearch.query.location.aggregation_result_extractor`:
```

Comment:
Tags 

*   `ezplatform.search.elasticsearch.query.content.aggregation_visitor`
*   `ezplatform.search.elasticsearch.query.content.aggregation_result_extractor`

needs to mentioned and described in documentation as well.